tabpy-server/server_tests/test_service_info_handler.py

```diff
@@ -0,0 +1,87 @@
+from tornado.testing import AsyncHTTPTestCase
+from tabpy_server.app import TabPyApp
+import simplejson as json
+from tabpy_server import __version__
+import tempfile, os
+
+
+def _create_expected_info_response(settings, tabpy_state):
+    response = {}
+    response['description'] = tabpy_state.get_description()
+    response['creation_time'] = tabpy_state.creation_time
+    response['state_path'] = settings['state_file_path']
+    response['server_version'] = settings['server_version']
+    response['name'] = tabpy_state.name
+    response['versions'] = settings['versions']
+    return response
+
+
+def _get_app_and_state(config_file=None):
+    tabpy_app = TabPyApp(config_file=config_file)
+    tornado_app = tabpy_app._create_tornado_web_app()
+
+    return tornado_app, tabpy_app.settings, tabpy_app.tabpy_state
+
+
+class TestServiceInfoHandlerDefault(AsyncHTTPTestCase):
+    def get_app(self):
+        app, self.settings, self.state = _get_app_and_state()
+        return app
+
+    def test_info(self):
+        """
+        Tests that /info returns the correct response on a vanilla TabPy instance.
+        """
+        response = self.fetch('/info')
+        self.assertEqual(response.code, 200)
+        actual_response = json.loads(response.body)
+        expected_response = _create_expected_info_response(self.settings, self.state)
+
+        self.assertDictEqual(actual_response, expected_response)
+
+
+class TestServiceInfoHandlerWithAuth(AsyncHTTPTestCase):
+    def get_app(self):
+        test_pwd_file = tempfile.NamedTemporaryFile(prefix='___TestServiceInfoHandlerWithAuth_', suffix='.txt', delete=False)
+        test_pwd_file.write(b'username password')
+        test_pwd_file.close()
+
+        test_config = tempfile.NamedTemporaryFile(prefix='___TestServiceInfoHandlerWithAuth_', suffix='.conf',
+                                                  delete=False)
+        config = '[TabPy]\n' \
+                 'TABPY_PWD_FILE = {}\n' \
+                 'TABPY_STATE_PATH = ./tabpy-server/tabpy_server'.format(test_pwd_file.name)
```

Comment:
We don't want to use actual `state.ini` as it may contain user data or something which may interfere with tests. Instead, create temp state file and set an absolute path to it with the parameter. An example in `test_pwd_file.py` for how passwords files are created and used.

Reply:
We shouldn't expect unit tests to pass on a deployed instance of TabPy.  The tests should use the expected (default) state.ini - or have a separate integration test for that if we decide to not test it here.